commit 78a678dfc0987c682ab7dd8e10759cc2761bb5f0 (refs/changes/98/2698/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-10-09T23:51:28+00:00 (2 years ago)
    
    TOOLS-1877 bin/crimport broken by Github API changes
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>
    Approved by: Joshua M. Clulow <jmc@joyent.com>

diff --git a/bin/crimport b/bin/crimport
index 39f8826..1608ba2 100755
--- a/bin/crimport
+++ b/bin/crimport
@@ -255,7 +255,6 @@ function igp_github_protect
 
 	local TOKEN="$(cat $igp_github_token_file)"
 	local BRANCH="master"
-	local PREVIEW_HEADER="Accept: application/vnd.github.loki-preview+json"
 	local UNIQUE="$RANDOM$RANDOM$RANDOM:"
 
 	# Check that the Gerrit user is able to push
@@ -280,7 +279,7 @@ function igp_github_protect
 	fi
 
 	# Set branch protection on master, so that only Gerrit can push.
-	curl -fsS -u "$TOKEN:x-oauth-basic" -H "$PREVIEW_HEADER" -X PUT -d@- \
+	curl -fsS -u "$TOKEN:x-oauth-basic" -X PUT -d@- \
 	    "$igp_github_api/repos/$1/$2/branches/$BRANCH/protection" \
 	    > /dev/null <<-EOF
 		{
@@ -288,6 +287,14 @@ function igp_github_protect
 		        "strict": true,
 		        "contexts": [ ]
 		    },
+		    "required_pull_request_reviews": {
+		        "dismissal_restrictions": {
+		            "users": [],
+		            "teams": []
+		        },
+		        "dismiss_stale_reviews": true,
+		        "require_code_owner_reviews": true
+		    },
 		    "enforce_admins": true,
 		    "restrictions": {
 		        "users": [
